src/Microsoft.Framework.Configuration/Internal/ConfigurationFocus.cs

```diff
@@ -72,5 +73,10 @@ public void Set(string key, string value)
         {
             return _root.GetConfigurationSections(_prefix + key);
         }
+
+        public void Reload()
+        {
+            throw new NotImplementedException();
```

Comment:
I think the interface design here is debatable. If the sub section doesn't provide the functionality of reload why does it provide such a method? Throw exception feels weird since it action shouldn't do harm to anything.